### PR TITLE
feat(authorization): add bearer token authorization

### PR DESCRIPTION
**Pull request by nithinssabu**
## Description of the change

[Describe your changes here]

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update

## Checklist

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I have opened this pull request against the `alpha` branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

[If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...]

### CODE REVIEW

**Comment by jwulf** [comment]
Hi @nithinssabu , thanks for the contribution. 

Can you give me an idea of the use case for this?

**Comment by nithinssabu** [comment]
> Hi @nithinssabu , thanks for the contribution.
> 
> Can you give me an idea of the use case for this?

Hey @jwulf 

This is required as part of a change in the deployment functionality within Web Modeler. The [epic](https://github.com/camunda/web-modeler/issues/10217) is about unifying the deployment experience between the SaaS and Self-Managed product contexts.

In SaaS, deployments are easy from Web Modeler. Users select a cluster and deploy. We want to bring the same experience also to Self-Managed (currently users fill out a manual entry form where they have to enter the cluster details including the credentials).

This change is relevant for the [authorization slice](https://github.com/camunda/web-modeler/issues/10572) of the epic where we will switch to using the user token instead of the M2M token generated using client ID and secret.

In short, it's about changing **from this**:
image
**to this:**
image


PS: I was planning to reach out to you today. I just wanted to try this out first to verify if it works. I will read the contribution guide and improve the PR, add tests, fill more details, etc. before marking it as ready. In the meantime, if you already find any red flags or possible improvements in the approach, please share.

**Review by jwulf — APPROVED** [review]
LGTM.
